tests/ui/simd/intrinsic/generic-gather-pass.rs

```diff
@@ -27,7 +31,7 @@ fn main() {
 
         let r_strided = simd_gather(default, pointers, mask);
 
-        assert_eq!(r_strided, s_strided);
+        assert_eq!(r_strided.to_array(), s_strided.to_array());
```

Comment:
macro-wise, should we replace this with `assert_eq_simd!` rather than calling `to_array` everywhere?